node/node-core/src/eth/mod.rs

```diff
@@ -0,0 +1,263 @@
+use crate::{
+    error::RpcError,
+    eth::{call::CallRequest, filter::Filter, transaction::EthTransactionRequest},
+    types::Index,
+};
+use ethers_core::{
+    abi::ethereum_types::H64,
+    types::{Address, BlockNumber, Bytes, Transaction, TxHash, H256, U256},
+};
+use serde::{de::DeserializeOwned, Deserialize, Deserializer, Serialize};
+
+pub mod block;
+pub mod call;
+pub mod filter;
+pub mod receipt;
+pub mod transaction;
+pub mod trie;
+pub mod utils;
+
+/// Represents ethereum JSON-RPC API
```

Comment:
Some RPC methods I'd love to see added
- `debug_traceTransaction` is a standard RPC method. This is a [great feature](https://hardhat.org/hardhat-network/#the-debug-tracetransaction-method) of Hardhat: many node providers don't support this opcode, but you can start hardhat against an archive node then use this
- Anvil RPC methods for all applicable cheat codes: `roll`, `warp`, `fee`, `store`, `etch`, `deal`, `pranks`, `setNonce`, etc. (probably missed some). Hardhat's convention is prefix all these with `hardhat_` so it's clear they're non-standard. So we can probably use `anvil_` as the prefix here

Reply:
Perhaps there would be a way to easily adapt existing hardhat tests to be able to run against anvil, like importing a hardhat plugin that alters things like the `hardhat` prefix in the custom RPCs to be `anvil`